### PR TITLE
Docs improvements

### PR DESCRIPTION
**Pull request by jhrcek**
Few minor improvements to user docs.

[ci skip]

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users have been recorded in the changelog (add file to `changelog.d` directory).
* [x] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: Cabal/doc/installing-packages.rst [hunk 1]
```diff
@@ -200,7 +200,7 @@ This is just syntactic sugar for
 ::
 
     repository hackage.haskell.org
-      url: hackage.haskell.org:http://hackage.haskell.org/packages/archive
+      url: http://hackage.haskell.org/packages/archive
```
Comment (jhrcek) [inline Cabal/doc/installing-packages.rst:203]: Not sure about this URL, but this seems like copy-paste error based on the above.

**Comment by phadej** [comment]
FWIW, they are not removed, they are just not in the `--help` output (duplicating `v2-` and prefixless variants)

File: Cabal/doc/nix-local-build.rst [hunk 2]
```diff
@@ -191,7 +191,7 @@ version of cabal-install:
    ``dist-newstyle/build/x86_64-linux/ghc-8.0.1/p-0.1/c/pexe/build/pexe/pexe``
    (you can see why we want this to be an implementation detail!)
 
-- In cabal-install-2.2 and above, the ``/c/`` part of the above path
```
Comment (jhrcek) [inline Cabal/doc/nix-local-build.rst:194]: This led to some bad formatting

Screenshot from 2020-06-30 14-17-05

File: Cabal/doc/nix-local-build.rst [hunk 3]
```diff
@@ -235,10 +235,10 @@ this folder (the most important two are first):
     a local package are unchanged, ``cabal v2-build`` will skip
     invoking ``setup build`` entirely (saving us from a possibly
     expensive call to ``ghc --make``). The full list of source files
-    participating in compilation are determined using
-    ``setup sdist --list-sources`` (thus, if you do not list all your
-    source files in a Cabal file, you may fail to recompile when you
-    edit them.)
+    participating in compilation is determined using
+    ``cabal sdist --list-only``. Thus if you do not list all your
```
Comment (jhrcek) [inline Cabal/doc/nix-local-build.rst:239]: Not sure about this, but this seems to be v2- equivalent?

File: Cabal/doc/nix-local-build.rst [hunk 4]
```diff
@@ -261,8 +261,4 @@ this folder (the most important two are first):
 
 
 Note that every package also has a local cache managed by the Cabal
-build system, e.g., in ``$distdir/cache``.
-
-There is another useful file in ``dist-newstyle/cache``,
```
Comment (phadej) [inline Cabal/doc/nix-local-build.rst:266]: unintentionally rmeoved `plan.json` paragraph

Comment (jhrcek) [inline Cabal/doc/nix-local-build.rst:266]: The plan.json is mentioned few lines above this, so this seems redundant.

File: Cabal/doc/nix-local-build.rst [hunk 5]
```diff
@@ -118,9 +118,9 @@ across projects. To be more precise:
 1. A **local package** is one that is listed explicitly in the
    ``packages``, ``optional-packages`` or ``extra-packages`` field of a
    project. Usually, these refer to packages whose source code lives
-   directly in a folder in your project (although, you can list an
+   directly in a folder in your project. But you can list an
    arbitrary Hackage package in ``extra-packages`` to force it to be
```
Comment (phadej) [inline Cabal/doc/nix-local-build.rst:122]: extra-packages could be a link (especially important as there it tells that feature is not implemented).

File: Cabal/doc/nix-local-build.rst [hunk 6]
```diff
@@ -118,9 +118,9 @@ across projects. To be more precise:
 1. A **local package** is one that is listed explicitly in the
    ``packages``, ``optional-packages`` or ``extra-packages`` field of a
    project. Usually, these refer to packages whose source code lives
-   directly in a folder in your project (although, you can list an
```
Comment (jhrcek) [inline Cabal/doc/nix-local-build.rst:121]: I find the notes in parenthesis everywhere a bit annoying (forcing me to context-switch :sweat_smile: , breaking the flow of text)

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -112,7 +112,7 @@ If none of these let you reproduce, there might be some race condition
 or continuous integration breakage; please file a bug.
 
 **Running tests locally.**
-To run tests locally with `new-build`, you will need to know the
+To run tests locally with `v2-build`, you will need to know the
```
Comment (phadej) [inline CONTRIBUTING.md:115]: should here be double ticks

**Comment by jhrcek** [comment]
The [ci skip] mentioned in https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md doesn't seem to skip ci :)

File: Cabal/doc/nix-local-build.rst (hunk 4 above)
Comment (jhrcek) [reply Cabal/doc/nix-local-build.rst:266]: Actually It was intentional, as that file is mentioned in more structured paragraph just above this removed one.

**Comment by phadej** [comment]
no, we should remove `[ci skip]` mentions. I'm not see it returning any time soon.

File: CONTRIBUTING.md (hunk 7 above)
Comment (jhrcek) [reply CONTRIBUTING.md:115]: I'll use double ticks here. But should I revert the new -> v2 changes in this doc to avoid unnecessary changes?

**Comment by phadej** [comment]
I'll check `CONTRIBUTING.md` myself. There is probably something else to improve / correct too.

File: Cabal/doc/nix-local-build.rst (hunk 3 above)
Comment (phadej) [reply Cabal/doc/nix-local-build.rst:239]: This is technically correct, and `cabal (v2-)sdist` command is something people can run easily, so is better.

File: Cabal/doc/installing-packages.rst (hunk 1 above)
Comment (phadej) [reply Cabal/doc/installing-packages.rst:203]: yes

File: Cabal/doc/nix-local-build.rst (hunk 4 above)
Comment (phadej) [reply Cabal/doc/nix-local-build.rst:266]: I see.

Comment (phadej) [reply Cabal/doc/nix-local-build.rst:266]: As you commented, if there are explanation above.

File: CONTRIBUTING.md (hunk 7 above)
Comment (phadej) [reply CONTRIBUTING.md:115]: No, using `v2-` is better.

Comment (jhrcek) [reply CONTRIBUTING.md:115]: Ok, keeping v2. Also a I realized this is markdown file, so single ticks should be the way to go.

File: Cabal/doc/nix-local-build.rst (hunk 5 above)
Comment (jhrcek) [reply Cabal/doc/nix-local-build.rst]: Added link to the second occurrence below. But did so blindly (my attempts to build the docs locally with python2 failed :( so please check the syntax.

**Comment by jhrcek** [comment]
Please don't merge yet, I'll add more fixes over the couple of next days as I finish reading through the user guide and let you know when I'm done.

**Comment by phadej** [comment]
@jhrcek is there still something?

**Comment by jhrcek** [comment]
@phadej by all means merge, if you find it useful as is. I'm still reading the guide (intending to read it all to get better idea about all the functionality), but I can create separate PR with the rest later.

**Comment by phadej** [comment]
You should be able to build the docs by

```
cd Cabal
make users-guide
```

That should setup virtualenv so "just work".

**Comment by phadej** [comment]
Merged as https://github.com/haskell/cabal/pull/6951

**Comment by jhrcek** [comment]
> cd Cabal
> make users-guide

Thanks, but using those steps don't fails for me with
```bash
$ make users-guide
...
mkdir -p dist/doc/users-guide
(. ./.python-sphinx-virtualenv/bin/activate && sphinx-build doc dist/doc/users-guide)
Running Sphinx v3.1.2
WARNING: Support for evaluating Python 2 syntax is deprecated and will be removed in Sphinx 4.0. Convert /home/jhrcek/Devel/github.com/haskell/cabal/Cabal/doc/conf.py to Python 3 syntax.

Configuration error:
There is a syntax error in your configuration file: invalid syntax (cabaldomain.py, line 153)

make: *** [Makefile:59: dist/doc/users-guide/index.html] Error 2
```
I use latest fedora workstation which doesn't ship with python2 out of the box. And I don't want to pollute my environment by installing python 2 :smile:  I guess I'll need to wait for [this migration to python3](https://github.com/haskell/cabal/pull/6754) to be berge to able to build the docs locally.

**Comment by phadej** [comment]
@jhrcek thanks for input. I'll try to fix that ASAP. Documentation is quite important.

**Pull request closed without merging**